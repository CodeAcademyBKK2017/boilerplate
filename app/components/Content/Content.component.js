import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Content.style.js';
import {
  Text
  , TextInput
  , View
} from 'react-native';

class Content extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Please type your note below</Text>
        <TextInput
          style={styles.textinput}
          underlineColorAndroid='transparent'
          multiline={true}
          onChangeText={this.props.onKeyPress}
        />
      </View>
    );
  }
}

Content.propTypes = {
  onKeyPress: PropTypes.func
};
Content.defaultProps = {
  onKeyPress: noop
};

export default Content;