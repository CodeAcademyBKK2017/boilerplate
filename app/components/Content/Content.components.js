import React, {Component} from 'react';
import {

  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

class Content extends Component {
  render () {
    return (
      <View style={styles.conView}>
        <Text style={styles.welcome}>
         Please type your note below
        </Text>
        <TextInput
          style={styles.textin}
          // onChangeText={this.state.nans}
          multiline = {true}
          underlineColorAndroid = 'transparent'
          numberOfLines = {4}   
          placeholder='-Understand how react-native works.'
        />
      </View>
    );
  }
}
const styles = StyleSheet.create({
  welcome: {
    fontSize: 12,
    // textAlign: 'center',
    margin: 10
   
  },
  conView: {
    flex: 1,
    marginTop: 10
  },
  textin: {
    // width: 300,
    // height: 300,
    borderColor: 'gray',
    borderWidth: 1,
    flex: 1,
    backgroundColor: '#fff'

  }
});
export default Content;