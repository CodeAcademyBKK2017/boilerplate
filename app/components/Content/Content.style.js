import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1
  },
  text: {
    fontStyle: 'italic'
  },
  textinput: {
    textAlignVertical: 'top',
    borderColor: 'gray',
    borderWidth: 1,
    padding: 5,
    flex: 1
  }
});