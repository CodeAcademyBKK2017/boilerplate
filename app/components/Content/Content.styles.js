import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    flex: 1,
    paddingBottom: 10
  },
  title: {
    fontSize: 13,
    color: '#444',
    fontStyle: 'italic'
  },
  textInput: {
    flex: 1,

    marginVertical: 10,
    paddingVertical: 15,
    paddingHorizontal: 12,

    borderWidth: 1,
    borderColor: '#c4c4c5',
    borderRadius: 3,
    
    backgroundColor: '#ffffff',
    textAlignVertical: 'top'
  }
});