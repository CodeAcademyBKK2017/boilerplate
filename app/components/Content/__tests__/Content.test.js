
import React from 'react';
import 'react-native';
import Content from '../Content.component';
import renderer from 'react-test-renderer';

describe('Content', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<Content />);
    expect(snapshot).toMatchSnapshot();
  });
});