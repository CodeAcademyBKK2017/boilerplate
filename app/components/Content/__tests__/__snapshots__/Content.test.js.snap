// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    Please type your note below
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
